### ISSUE TITLE
replacement for valid-typeof rule

### ISSUE THREAD

**Issue by sundartinesh** [eslint/eslint#11140]
**What rule do you want to change?**
**valid-typeof**

**Does this change cause the rule to produce more or fewer warnings?**
**this is not showing any errors in my repo**

**How will the change be implemented? (New option, new default behavior, etc.)?**
**I have wriiten a new rule as a replacement**

**Please provide some example code that this change will affect:**

```js
typeof a === null;
typeof b === undefined;
```

**What does the rule currently do for this code?**
**the rule must show error,but it is not showing any errors**

**What will the rule do after it's changed?**
**I have written a rule that will show errors**

**Are you willing to submit a pull request to implement this change?**
**absolutely yes**

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue. Have you tried using the `requireStringLiterals` option for `valid-typeof`? With that option enabled, the rule will report errors for your example code.
